lldb/tools/lldb-dap/Request/Request.h

```diff
@@ -0,0 +1,45 @@
+//===-- Request.h ---------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLDB_TOOLS_LLDB_DAP_REQUEST_REQUEST_H
+#define LLDB_TOOLS_LLDB_DAP_REQUEST_REQUEST_H
+
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/JSON.h"
+
+namespace lldb_dap {
+struct DAP;
+
+class Request {
```

Comment:
As far as naming goes, What do you think of `RequestHandler`? That makes this type more distinct from the `Request` which is a specific thing in the DAP.

Then the handlers could be named like `AttachHandler` or more generically `<command>Handler`.


Should we make these types move only?

e.g.

```
  // No copy/move
  Request(Request &&) = delete;
  Request &operator=(Request &&) = delete;
  Request(const Request &) = delete;
  Request &operator=(const Request &) = delete;
```